api-admin/src/main/java/bio/terra/pearl/api/admin/service/admin/AdminUserExtService.java

```diff
@@ -71,39 +82,126 @@ public AdminUser get(UUID id, String portalShortcode, AdminUser operator) {
     return adminUser;
   }
 
-  public List<AdminUser> getAll(AdminUser operator) {
-    if (operator.isSuperuser()) {
-      return adminUserService.findAllWithRoles();
-    }
-    throw new PermissionDeniedException(
-        "User %s does not have permissions to list all users".formatted(operator.getUsername()));
+  @SuperuserOnly
+  public List<AdminUser> getAll(OperatorAuthContext authContext) {
+    return adminUserService.findAllWithRoles();
   }
 
-  public List<AdminUser> findByPortal(String portalShortcode, AdminUser operator) {
-    Portal portal = authUtilService.authUserToPortal(operator, portalShortcode);
-    return adminUserService.findAllWithRolesByPortal(portal.getId());
+  @EnforcePortalPermission(permission = "BASE")
+  public List<AdminUser> findByPortal(PortalAuthContext authContext) {
+    return adminUserService.findAllWithRolesByPortal(authContext.getPortal().getId());
   }
 
-  public AdminUser create(NewAdminUser newUserParams, AdminUser operator) {
-    if (newUserParams.superuser && !operator.isSuperuser()) {
-      throw new PermissionDeniedException(
-          "User %s does not have permissions to create superusers"
-              .formatted(operator.getUsername()));
+  @SuperuserOnly
+  public AdminUser createSuperuser(OperatorAuthContext authContext, String username) {
+    AdminUser newUser = AdminUser.builder().username(username).superuser(true).build();
+    DataAuditInfo auditInfo =
+        DataAuditInfo.builder().responsibleAdminUserId(authContext.getOperator().getId()).build();
+    AdminUser adminUser = adminUserService.create(newUser, auditInfo);
+    adminEmailService.sendWelcomeEmail(null, adminUser);
+    return adminUser;
+  }
+
+  @EnforcePortalPermission(permission = "admin_user_edit")
+  @Transactional
+  public AdminUser createAdminUser(
+      PortalAuthContext authContext, String username, List<String> roleNames) {
+    authorizeRoles(authContext, roleNames);
+    AdminUser newUser = AdminUser.builder().username(username).superuser(false).build();
+    newUser
+        .getPortalAdminUsers()
+        .add(PortalAdminUser.builder().portalId(authContext.getPortal().getId()).build());
+    DataAuditInfo auditInfo =
+        DataAuditInfo.builder().responsibleAdminUserId(authContext.getOperator().getId()).build();
+    AdminUser adminUser = adminUserService.create(newUser, auditInfo);
```

Comment:
We probably want to check for conflicts in the underlying create here, right now this throws a SQL exception